lib/src/solver/report.dart

```diff
@@ -375,32 +412,57 @@ $contentHashesDocumentationUrl
         maxAge: Duration(days: 3),
       );
 
+      var messagePrefix = '(';
+      var hasAdvisory = false;
+      if (status.advisoriesUpdated != null) {
+        final advisories = await id.source.getAdvisoriesForPackageVersion(
+          id,
+          _cache,
+          Duration(days: 3),
+        );
+        hasAdvisory = advisories != null && advisories.isNotEmpty;
+        if (hasAdvisory) {
+          final advisoryFootnotes = <int>[];
```

Comment:
This list will always contain [start, start+1...., end] Perhaps all you need to pass to `_constructAdvisoriesMessage` is the start and end index.

Reply:
I will keep this, since as a follow up I would like to to an optimization where footnotes can be reused between dependencies. In that case the assumption that the footnotes are consecutive is not correct.